### PR TITLE
feat: New Queued resources Samples: Create, Create Spot, Get, Delete operations + tests

### PR DESCRIPTION
**Pull request by Thoughtseize1**
### 📑Created a full set of Queued Resources Samples:
| Region Tag | File | 
| ----- | -----| 
| tpu_queued_resources_create  |  queued_resources_create.py |
| tpu_queued_resources_network | queued_resources_create_network.py  |
| tpu_queued_resources_create_spot  | queued_resources_create_spot.py  |
| tpu_queued_resources_startup_script  |  queued_resources_create_startup_script.py |
| tpu_queued_resources_time_bound  |  queued_resources_create_time_bound.py |
| tpu_queued_resources_get |  queued_resources_get.py |
| tpu_queued_resources_list  |  queued_resources_list.py |
| tpu_queued_resource_delete | queued_resources_delete.py  |
| tpu_queued_resources_delete_force |  queued_resources_delete_force.py |

### ❗Issue with Deleting operation in `google.cloud.tpu_v2alpha1`

The delete operation in `queued_resources_delete.py` have issue [here](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/6b63f43bebc980070fdf7c9d9b437d4feef2a227/tpu/queued_resources_delete.py#L32-L40) with this 
block of code.
```
    try:
        op = client.delete_queued_resource(request=request)
        op.result()
        print(f"Queued resource '{queued_resource_name}' successfully deleted.")
    except TypeError as e:
        print(f"Error deleting resource: {e}")
        print(f"Queued resource '{queued_resource_name}' successfully deleted.")
```
**Issues Encountered:**
Resource Successfully Deleted but we got TypeError: After invoking op.result() the resource is indeed successfully deleted. However, despite the deletion being successful, the function throws a `TypeError: Could not convert Any to QueuedResource.`
This error suggests that the API response could not be properly deserialized or converted into a QueuedResource object, even though the deletion itself completed as expected.

The code should not throw a TypeError when the resources are successfully deleted. We cannot avoid using op.result() because it guarantees that an Exception will be raised when attempting to delete a resource in a status where deletion is not possible. This behavior ensures proper error handling in situations where the resource cannot be deleted, but the current implementation incorrectly throws a TypeError even when deletion is successful.

This is precisely why I had to use two separate except blocks in the code.

## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md)
- [ ] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#readme-file)
- [x] **Tests** pass:   `nox -s py-3.9` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [x] **Lint** pass:   `nox -s lint` (see [Test Environment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: tpu/queued_resources_delete.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    try:
+        op = client.delete_queued_resource(name=name)
+        op.result()
+        print(f"Queued resource '{queued_resource_name}' successfully deleted.")
+    except TypeError as e:
+        print(f"Error deleting resource: {e}")
+        print(f"Queued resource '{queued_resource_name}' successfully deleted.")
+    except Exception as e:
+        print(f"Error deleting resource: {e}")
```
Comment (m-strzelczyk) [inline tpu/queued_resources_delete.py:41]: I think in this case we can `raise e`, so we don't ignore proper issues.

File: tpu/test_queued_resources.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        if resource.state.state in STATUSES:
+            try:
+                print(f"Attempting to delete resource '{RESOURCE_NAME}'...")
+                queued_resources_delete.delete_queued_resources(
+                    PROJECT_ID, ZONE, RESOURCE_NAME
+                )
+                print("Resource and TPU successfully deleted. Exiting...")
+                return True
+            except Exception:
+                continue
```
Comment (m-strzelczyk) [inline tpu/test_queued_resources.py:59]: Maybe print the error, so it's easier to debug any potential issues.

File: tpu/test_queued_resources.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        print("Resource is not in a deletable state. Waiting...")
+        time.sleep(60)
+        try:
+            node = get_tpu.get_cloud_tpu(PROJECT_ID, ZONE, TPU_NAME)
+            print("NODE:", node.name)
+            if node and node.state == Node.State.READY:
+                print(f"Attempting to delete TPU '{TPU_NAME}'...")
+                delete_tpu.delete_cloud_tpu(PROJECT_ID, ZONE, TPU_NAME)
+        except Exception:
+            continue
```
Comment (m-strzelczyk) [inline tpu/test_queued_resources.py:69]: print

File: tpu/test_queued_resources.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+STATUSES = [
+    States.State.ACCEPTED,
+    States.State.WAITING_FOR_RESOURCES,
+    States.State.SUSPENDED,
+    States.State.FAILED,
+]
+
+
+def clean_resource() -> None:
```
Comment (m-strzelczyk) [inline tpu/test_queued_resources.py:45]: Add some comment why you also delete TPU here, that it's liked to the queued resource etc.

**Review by m-strzelczyk — CHANGES_REQUESTED** [review]

File: tpu/queued_resources_delete.py (hunk 1 above)
Comment (Thoughtseize1) [reply tpu/queued_resources_delete.py]: I simplified this code block.

File: tpu/test_queued_resources.py (hunk 3 above)
Comment (Thoughtseize1) [reply tpu/test_queued_resources.py]: Done

File: tpu/test_queued_resources.py (hunk 2 above)
Comment (Thoughtseize1) [reply tpu/test_queued_resources.py]: Done

**Review by m-strzelczyk — APPROVED** [review]

**Pull request merged**